test/extensions/network/dns_resolver/cares/dns_impl_test.cc

```diff
@@ -663,10 +665,30 @@ TEST_F(DnsImplConstructor, BadCustomResolvers) {
       EnvoyException, "DNS resolver 'foo' is not an IP address");
 }
 
-class DnsImplTest : public testing::TestWithParam<Address::IpVersion> {
+std::vector<std::tuple<Address::IpVersion, bool>> paramGenerator() {
```

Comment:
I don't think we need to parameterize every test. Can you just fix all the tests for the new behavior and then add a specific test with a reverted scoped runtime to make sure the revert behavior works also?